lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -473,14 +481,22 @@ func toModelsNetworkConfig(cc types.ContainerCreateConfig) *models.NetworkConfig
 	return nc
 }
 
-//This function is used to turn any call from docker create -v <stuff> into a volumeFields object.
-//the -v has 3 forms. 1: -v <anonymous mount path>, -v <Volume Name>:<Destination Mount Path>, and -v <Volume Name>:<Destination Mount Path>:<mount flags>
+// processVolumeParam is used to turn any call from docker create -v <stuff> into a volumeFields object.
+// The -v has 3 forms. -v <anonymous mount path>, -v <Volume Name>:<Destination Mount Path> and
+// -v <Volume Name>:<Destination Mount Path>:<mount flags>
 func processVolumeParam(volString string) (volumeFields, error) {
 	volumeStrings := strings.Split(volString, ":")
 	fields := volumeFields{}
 
-	//This switch determines which type of -v was invoked.
-	switch len(volumeStrings) {
+	// Error out if the intended volume is a directory on the client filesystem.
+	volStrLen := len(volumeStrings)
+	if volStrLen > 1 && strings.HasPrefix(volumeStrings[0], "/") {
+		invalidVolumeErr := fmt.Errorf("%s does not support mounting directories as a data volume.", ProductName())
```

Comment:
This smells like it should be its own type that satisfies the `error` interface which we can then type assert.